From e2dfa761102bac39267d2e60c8403348a0c9a2e5 Mon Sep 17 00:00:00 2001
From: Dave Yeo <dave.r.yeo@gmail.com>
Date: Sat, 20 Dec 2014 19:44:34 -0800
Subject: [PATCH 6/6] [Sydneyaudio] Stub out a function that depends on
 libcubex

Stub out the function that queries the number of channels that the hardware can use and return 2.
---
 content/media/AudioStream.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/content/media/AudioStream.cpp b/content/media/AudioStream.cpp
index b457ebb..99b27ac 100644
--- a/content/media/AudioStream.cpp
+++ b/content/media/AudioStream.cpp
@@ -756,6 +756,7 @@ int AudioStream::MaxNumberOfChannels()
 {
   uint32_t maxNumberOfChannels, rv;
 
+#ifndef XP_OS2
   rv = cubeb_get_max_channel_count(GetCubebContext(), &maxNumberOfChannels);
 
   if (rv != CUBEB_OK) {
@@ -763,6 +764,9 @@ int AudioStream::MaxNumberOfChannels()
   }
 
   return static_cast<int>(maxNumberOfChannels);
+#else
+  return 2;
+#endif
 }
 
 static void SetUint16LE(uint8_t* aDest, uint16_t aValue)
-- 
2.0.0

